src/test/java/com/datatorrent/netlet/DefaultEventLoopTest.java

```diff
@@ -300,4 +300,88 @@ public void registered(SelectionKey key)
     assertFalse(client2.isConnected());
   }
 
+  @Test
+  public void startStop() throws Exception
+  {
+    assertTrue(defaultEventLoop.isActive());
+    assertEquals(thread, defaultEventLoop.start());
+    defaultEventLoop.stop();
+    assertTrue(defaultEventLoop.isActive());
+    defaultEventLoop.stop();
+    thread.join();
+    assertFalse(defaultEventLoop.isActive());
+  }
+
+  @Test
+  public void startWithExtraStop() throws Exception
+  {
+    assertTrue(defaultEventLoop.isActive());
+    defaultEventLoop.stop();
+    thread.join();
```

Comment:
It is not good for tests to hang indefinitely. You should have a failsafe timeout value that is usually large enough that it doesn't lead to a false negative and when that timeout occurs you do fail the test.